modules/client/src/internal/client/methods.ts

```diff
@@ -625,4 +637,75 @@ export class ClientMethods extends ClientCore implements IClientMethods {
       ),
     );
   }
+
+  /**
+   * Retrieves the list of plugins available on the PluginRegistry
+   *
+   * @param {PluginQueryParams} {
+   *     limit = 10,
+   *     skip = 0,
+   *     direction = SortDirection.ASC,
+   *     sortBy = PluginSortBy.SUBDOMAIN,
+   *     subdomain
+   *   }
+   * @return {*}  {(Promise<PluginRepo[] | null>)}
+   * @memberof ClientMethods
+   */
+  public async getPlugins({
+    limit = 10,
+    skip = 0,
+    direction = SortDirection.ASC,
+    sortBy = PluginSortBy.SUBDOMAIN,
+    subdomain,
+  }: PluginQueryParams = {}): Promise<PluginRepo[]> {
+    let where = {};
+    if (subdomain) {
+      where = { subdomain_contains_nocase: subdomain };
+    }
+    const query = QueryPlugins;
+    const params = {
+      where,
+      limit,
+      skip,
+      direction,
+      sortBy,
+    };
+    const name = "plugin repos";
+    type T = { pluginRepos: SubgraphPluginRepoListItem[] };
+    const { pluginRepos } = await this.graphql.request<T>({
+      query,
+      params,
+      name,
+    });
+    return Promise.all(
+      pluginRepos.map(
+        async (pluginRepo: SubgraphPluginRepoListItem): Promise<PluginRepo> => {
+          let pluginRepoReleases: PluginRepoRelease[] = [];
+          for (const release of pluginRepo.releases) {
+            let metadata: PluginRepoReleaseMetadata;
+            if (!release.metadata) {
+              metadata = EMPTY_RELEASE_METADATA_LINK;
+            } else {
+              try {
+                const metadataCid = resolveIpfsCid(release.metadata);
+                const stringMetadata = await this.ipfs.fetchString(metadataCid);
+                const resolvedMetadata = JSON.parse(stringMetadata);
+                metadata = resolvedMetadata;
```

Comment:
Do we want to fetch every release's metadata? Feature-wise it makes total sense, but it also becomes so easy to stage a DOS attack if you pin 20 releases of a dummy plugin with an IPFS CiD pointing to a 50Gb file

Reply:
I mean this could be done literally with anything that has metadata, any proposal or dao has the same issue because we fetch the metadata. So if we want to protect against it i'd do it in the ipfs module, adding a size limit or similar

Reply:
The point is that we can only control our IPFS node, but nothing prevents someone from using Pinata and pinning a 1G file and setting it as the metadata everywhere.

We could limit the amount of bytes that our IPFS proxy will accept but independently, I would only fetch the metadata of the very last release and build on `getPlugins()` and then add a separate `getPlugin()` function, which attempts to fetch more data, only for the given plugin

Reply:
No, I was refering to the IPFS module in the SDK, the client can know how much data has downloaded for an IPFS request, if it exceeds lets say 100KB, we throw and return the default metadata

Reply:
This would require introducing the AbortController and rework the client so we can cancel after having downloaded just 100kb, not after the whole download

Anyway, given the new implementation, I think this can be resolved

Reply:
Agree. Anyway, I would address this on the IPFS proxy side, rather than on the client